File: simple.cpp
#include <iostream>
#include <fstream>
#include <unordered_map>

// Essentially uses the same approach as the python one
class measurement {
public:
    double min;
    double max;
    uint64_t count;
    double total;
};

int main() {
    std::unordered_map<std::string, measurement> measurements;
    std::string line;
    // Yes, still hardcoded
    std::ifstream file(R"(D:\dev\1brc\measurements.txt)");
    while (getline(file, line) ) {
        // line format: "name;value"
        auto pos = line.find(';');
        double value = std::stod(line.substr(pos + 1));
        auto name = line.substr(0, pos);
        if(!measurements.contains(name)) {
            measurements[name] = measurement{value, value, 1, value};
        }else{
            auto& m = measurements[name];
            m.min = std::min(m.min, value);
            m.max = std::max(m.max, value);
            m.count++;
            m.total += value;
        }
    }
    // I was too lazy to add sorting as required for the original challenge
    for (const auto& [name, m] : measurements) {
        std::cout << name << " min: " << m.min << " max: " << m.max << " avg: " << m.total / m.count << "\n";
    }
}

File: simple.py
from collections import defaultdict

# I hardcoded the path to the measurements
# As I am really really lazy.

def lines():
    with open('D:\\dev\\1brc\\measurements.txt', 'r', encoding='utf8') as f:
        for line in f:
            yield line


class Measurement:
    def __init__(self):
        self.count = 0
        self.total = 0
        self.min = float('inf')
        self.max = float('-inf')

    def add(self, value):
        self.count += 1
        self.total += value
        self.min = min(self.min, value)
        self.max = max(self.max, value)


results = defaultdict(Measurement)

for line in lines():
    location, measurement = line.split(";")
    measurement = float(measurement)
    results[location].add(measurement)

# To fix: This breaks for encoding reasons
for location, measurement in results.items():
    print(
        f"'{location}'='{measurement.min:.1f}/{(measurement.total / measurement.count) if measurement.count != 0 else 0:.1f}/{measurement.max:.1f}'",
        end="\n",
    )
